### DISCUSSION TITLE
Displaying a page within another

### DISCUSSION THREAD

**Discussion by Dani-Euronimus** [AvaloniaUI/Avalonia#14650] [Q&A] [answered]
Good afternoon,
I'm studying Avalonia and I'm having trouble displaying pages in the window. In my project, I have files: 
MainWindow.axaml:
`<views:MainView />`
MainView.axaml:
`<StackPanel>
	<TransitioningContentControl Content="{Binding CurrentPage}"/>
</StackPanel>`
MainViewModel.cs:
`public ViewModelBase CurrentPage => new HomePageViewModel(); `
HomePageView.axaml:
`<StackPanel>
	<TextBox Text="{Binding Greeting}"/>
</StackPanel>`
HomePageViewModel.cs:
`public class HomePageViewModel : ViewModelBase
{
    public string Greeting => "text! ";
}`
When running in debug mode, I see such a window:
![avalonia_test1](https://github.com/AvaloniaUI/Avalonia/assets/36798386/bfcfe448-63d1-4652-84f1-cc0b57cc6260)

If I change the code in MainViewModel.cs:
`public UserControl CurrentPage => new HomePageView(); `
then the window displays the HomePageView.axaml page, but the TextBox field is empty:
![avalonia_test1_](https://github.com/AvaloniaUI/Avalonia/assets/36798386/f54b3a9a-20a5-491c-a153-c791045d88d9)
But it looks normal in the preview window:
![HomePage_designer](https://github.com/AvaloniaUI/Avalonia/assets/36798386/7c5a2d3d-9ee8-4422-a851-fcb01d438d77)

I tried to display the HomePageView.axaml page in MainWindow.axaml: <views:HomePageView/> and saw an empty TextBox. I suppose that for some reason, HomePageView cannot get data from HomePageViewModel.
Please tell me what I'm doing wrong, I've been searching for a solution for 3 days already.

**Comment by timunie** [comment, marked as answer]
You need to define it as DataTemplate. See: https://github.com/AvaloniaUI/Avalonia.Samples?tab=readme-ov-file#%EF%B8%8F-datatemplate-samples
